File: 단계별로 풀어보기/Step 08/2292.cpp
#include <iostream>
using namespace std;

int main() {
    int num;
    cin >> num;

    int cnt;
    if(num==1) cnt = 1;
    else {
        int x = (num-2)/6;
        int idx = 2;
        while(x<(idx-1)*(idx-2)/2 || idx*(idx-1)/2-1<x) { idx++; }
        cnt = idx;
    }

    cout << cnt;
    return 0;
}

File: StepByStep/Python/Step_07_기본 수학 1/2292.py
N = int(input())
if N == 1:
    print(1)
else:
    idx = [0]
    i = 1
    while True:
        idx.append(idx[i-1]+i)
        if N < 6*idx[i]+2:
            break
        else:
            i += 1
    print(i+1)

# rooms = 1
# cnt = 1
# while N > rooms:
#     rooms += 6*cnt
#     cnt += 1
# print(cnt)